drivers/modem/modem_cellular.c

```diff
@@ -350,6 +349,7 @@ static void modem_cellular_chat_on_csq(struct modem_chat *chat, char **argv, uin
 	rssi = atoi(argv[1]);
 
 	if (rssi == 99) {
+		data->rssi = INT16_MAX;
```

Comment:
Maybe INT16_MIN is more clear as a "reserved number" to signify the RSSI is non-existent?

Reply:
I changed the +CSQ handler to simply store the raw falue (0-31 and 99) to be parsed just before returning it to the user with the new function `modem_cellular_csq_parse_rssi`, this way we don't need to choose a reserved value, we can just use 99 like the CSQ does :)